### DISCUSSION TITLE
Problem with KAM calculations for hexagonal grids

### DISCUSSION THREAD

**Discussion by MarcoWitte1** [mtex-toolbox/mtex#631] [Announcements]
Hi Ralf,

with MTEX 5.5.0 I calculated KAM values of a hexagonal ebsd grid with:

KAM_test = KAM(ebsd,'threshold',4*degree,'order',2)./degree;

The results are mostly NaN with some values on a diagonal trace:

![mtex550_EBSD_KAM](https://user-images.githubusercontent.com/15085129/102188102-2ecb2500-3eb5-11eb-820b-9730233bd21c.png)

Any ideas what went wrong there?

Viele Grüße
Marco

**Comment by kilir** [comment]
Hi Marco,
using the some of the hexgrid test data, I can't observe the interesting beam you have.

```
mtexdata ferrite
[grains, ebsd.grainId] = ebsd.calcGrains;
KAM_test = KAM(ebsd,'threshold',4*degree,'order',2)./degree;
plot(ebsd,KAM_test)
```
Do you have any further information on the data and on what you did to it?

Cheers,
Rüdiger

**Reply by MarcoWitte1** [reply]
Hi Rüdiger,

it seems to me, the problem is caused because I have two phases in my EBSD dataset.
If i use the whole gridified dataset I get this weird beam in the KAM data and mostly NaN
When I use only one phase with ebsd('1') then the data is not in a grid anymore and the KAM calculation works fine, but slow.
And with ebsd('1').gridify I get this error message:

Array indices must be positive integers or logical values.

Error in rotation/subsref (line 6)
    r.a = subsref(r.a,s(1));

Error in EBSD/project2FundamentalRegion (line 12)
  refRot = ebsd.rotations(accumarray(ebsd.grainId(isIndexed),find(isIndexed),[],@max));

Error in EBSDhex/KAM (line 32)
ebsd = ebsd.project2FundamentalRegion;

Cheers,
Marco

**Reply by kilir** [reply]
Hi Marco,
does this problem still exist? I could reproduce the error it with some self-made data a week ago; however, magically I don't encounter it anymore since I have lost that data. If you still have the issue and data which may provoke this, maybe you can send a sample of this dataset so it might be easier to reproduce?
Cheers,
Rüdiger

**Comment by MarcoWitte1** [comment]
Hi Rüdiger,

thanks for your reply! I wish you a happy new year!
Here is an example ebsd subset which leads to the described problem: 
[ebsd_test.zip](https://github.com/mtex-toolbox/mtex/files/5769587/ebsd_test.zip)

Cheers
Marco

**Comment by ralfHielscher** [comment]
Hi Marco, this is a bug in MTEX 5.5. I have fixed it in 753b5f1. Thank you for reporting it. The fix will be released together with MTEX 5.6 within the next days.

Ralf.

**Reply by MarcoWitte1** [reply]
Hi Ralf,

thank you! Now the KAM calculation works fine also for hexagonal grids. 
Only ebsd = ebsd.project2FundamentalRegion in @EBSDhex\KAM.m line 32 caused still the error:

Array indices must be positive integers or logical values.

Error in rotation/subsref (line 6)
    r.a = subsref(r.a,s(1));

Error in EBSD/project2FundamentalRegion (line 12)
  refRot = ebsd.rotations(accumarray(ebsd.grainId(isIndexed),find(isIndexed),[],@max));

But it worked after I commented this bit out.

Cheers
Marco

**Reply by ralfHielscher** [reply]
Hi Marco,

this issue had been fixed in 8f5df42 and went into MTEX 5.6.0 as well.

Ralf.
